models/migrations/v67.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/util"
+
+	"github.com/go-xorm/xorm"
+)
+
+func addPullRequestRebaseWithMergeCommit(x *xorm.Engine) error {
+	// RepoUnit describes all units of a repository
+	type RepoUnit struct {
+		ID          int64
+		RepoID      int64                  `xorm:"INDEX(s)"`
+		Type        int                    `xorm:"INDEX(s)"`
+		Config      map[string]interface{} `xorm:"JSON"`
+		CreatedUnix util.TimeStamp         `xorm:"INDEX CREATED"`
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	//Updating existing issue units
+	units := make([]*RepoUnit, 0, 100)
+	if err := sess.Where("`type` = ?", V16UnitTypePRs).Find(&units); err != nil {
+		return fmt.Errorf("Query repo units: %v", err)
+	}
+	for _, unit := range units {
+		if unit.Config == nil {
+			unit.Config = make(map[string]interface{})
+		}
+		if _, ok := unit.Config["AllowRebaseMergeCommit"]; !ok {
```

Comment:
I think `AllowRebaseMergeCommit` should be enabled only if `AllowMerge` is already enabled

Reply:
It might even make sense to not enable it at all for existing repositories, as doing so means interfering with the users choosen preferences without giving them the option to skip enabling it.

Reply:
Changed the migration, `AllowRebaseMerge` is now only set to `true` if all current merge styles are enabled.